File: frontend/src/types/staff.ts
import { QueueEntryStatus } from './customer';

export interface Organization {
  id: number;
  name: string;
  code: string;
  description?: string;
  created_at: string;
}

export interface StaffDashboardSummary {
  active_queues_count: number;
  waiting_customers_count: number;
  currently_serving_count: number;
  completed_today_count: number;
  average_wait_time: number;
  active_counters_count: number;
}

export interface NextCustomer {
  id: number;
  token_number: string;
  user_name: string;
  joined_at: string;
  estimated_wait_minutes: number;
  position: number;
}

export interface StaffQueue {
  id: number;
  organization_id?: number;
  organization_name?: string;
  location_id: number;
  location_name: string;
  service_id: number;
  service_name: string;
  status: string;
  current_token: number;
  now_serving_id?: number;
  now_serving_token?: string;
  now_serving_user?: string;
  now_serving_counter?: string;
  now_serving_status?: string;
  waiting_count: number;
  next_customers: NextCustomer[];
}

export interface StaffActionResponse {
  message: string;
  queue_entry_id: number;
  token_number: string;
  status: QueueEntryStatus;
  counter_name?: string;
}

File: backend/app/schemas/staff.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field
from app.models.enums import QueueEntryStatus

class StaffDashboardSummary(BaseModel):
    active_queues_count: int = 0
    waiting_customers_count: int = 0
    currently_serving_count: int = 0
    completed_today_count: int = 0
    average_wait_time: int = 0
    active_counters_count: int = 0

class NextCustomerOut(BaseModel):
    id: int
    token_number: str
    user_name: str
    joined_at: datetime
    estimated_wait_minutes: int
    position: int

class StaffQueueOut(BaseModel):
    id: int
    location_id: int
    location_name: str
    service_id: int
    service_name: str
    status: str
    current_token: int
    now_serving_id: Optional[int] = None
    now_serving_token: Optional[str] = None
    now_serving_user: Optional[str] = None
    now_serving_counter: Optional[str] = None
    now_serving_status: Optional[str] = None
    waiting_count: int = 0
    next_customers: List[NextCustomerOut] = []

class StaffActionResponse(BaseModel):
    message: str
    queue_entry_id: int
    token_number: str
    status: QueueEntryStatus
    counter_name: Optional[str] = None
